server/dex/dex.go

```diff
@@ -76,6 +113,13 @@ func (dm *DEX) Stop() {
 		log.Infof("Shutting down asset %q backend...", asset.Symbol)
 		asset.Backend.WaitForShutdown()
 	}
+	log.Infof("Stopping subsystems...")
+	for _, ssw := range dm.stopWaiters {
+		ssw.Stop()
+	}
+	for _, ssw := range dm.stopWaiters {
+		ssw.WaitForShutdown()
+	}
```

Comment:
This is quick and dirty, but there is likely to be a more appropriate order to the shutdown sequence.